Implement MAX1704x battery gauge (#962)
nanoframework/nanoFramework.IoT.Device: devices/MAX1704x/README.MD

> # MAX1704x battery gauge

Reviewer: Let's be explicit on the supported elements also, you need a new line after a title (MArkdown rule)

```suggestion
# MAX1704x/MAX17043/MAX17044/MAX17048/MAX17049 - Battery gauge

```

Author: done

Library designed for controlling MAX1704x IC.

> ### Functions

Reviewer: ```suggestion
## Functions

```

> The binding supports all funcionalities for MAX1704x family. Implemented based on https://github.com/sparkfun/SparkFun_MAX1704x_Fuel_Gauge_Arduino_Library/blob/main/src/SparkFun_MAX1704x_Fuel_Gauge_Arduino_Library.cpp

Reviewer: ```suggestion
The binding supports all functionalities for MAX1704x family. Implemented based on [SparkFun implementation](https://github.com/sparkfun/SparkFun_MAX1704x_Fuel_Gauge_Arduino_Library/blob/main/src/SparkFun_MAX1704x_Fuel_Gauge_Arduino_Library.cpp).
```

> ### Classes

Reviewer: In Markdown, like in Word, titles should be used in order of level, it's not good to have a level 3 without level 2. And the missing line after.

```suggestion
## Classes

```

* Max17043 - class for MAX17043
* Max17044 - class for MAX17044
* Max17048 - class for MAX17048
* Max17049 - class for MAX17049
* Max1704x - base class for all max IC

## Usage

> 

Reviewer: Can you copy/paste the usage we have in other I2C devices for ESP32 to setup the pins?
IT's a standard thing we've been doing to help in such cases.

```csharp
// Make sure your pins are configured correctly especially for ESp32
var i2cDevice = I2cDevice.Create(new I2cConnectionSettings(1, Max1704X.DefaultAddress));
var max = new Max17048(i2cDevice);
Console.WriteLine($"Voltage: {max.BatteryVoltage}");
Console.WriteLine($"Percent: {max.BatteryPercent} %");
```

> **Important**: make sure you properly setup the PWM pins especially for ESP32 before creating the `I2cDevice`.

Reviewer: it's not PWM :-D

Author: Fixed
